src/resolvers/git.cr

```diff
@@ -100,12 +100,25 @@ module Shards
       "git"
     end
 
+    private KEY_HOSTS = {"www.github.com", "github.com", "www.bitbucket.com", "bitbucket.com", "www.gitlab.com", "gitlab.com"}
+
     def self.normalize_key_source(key : String, source : String) : {String, String}
       case key
       when "git"
-        {"git", source}
+        uri = URI.parse(source)
+        downcased_host = uri.host.try &.downcase
+        scheme = uri.scheme.try &.downcase
+        if scheme.in?("git", "http", "https") && downcased_host && downcased_host.in?(KEY_HOSTS)
+          uri.scheme = "https"
+          downcased_path = uri.path.downcase
+          uri.path = downcased_path.ends_with?(".git") ? downcased_path : "#{downcased_path}.git"
+          uri.host = downcased_host.gsub("www.", "")
```

Comment:
This is more efficient:
```suggestion
          uri.host = downcased_host.lstrip("www.")
```

Reply:
It should actually be `lchop`. I have no idea why `lstrip` exists, it looks pretty dangerous! We should remove that immediately from the language. Same with `String#delete(string)`.

Reply:
(says the guy who seems to have implemented this... ouch!)

Reply:
Argh, yeah I've tripped over this before 😞